dotCMS/src/main/java/com/dotmarketing/filters/CMSFilter.java

```diff
@@ -54,8 +54,22 @@ public void init(FilterConfig arg0) throws ServletException {
 
     }
 
+    
     public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
             throws IOException, ServletException {
+        
+        try {
+            _doFilter(req,res,chain);
+        }
+        finally {
+            DbConnectionFactory.closeSilently();
+        }
+        
+    
+    }
+    
+    private void _doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 35 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This method has 95 lines, which is greater than the 75 lines authorized. Split it into smaller methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS138)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)